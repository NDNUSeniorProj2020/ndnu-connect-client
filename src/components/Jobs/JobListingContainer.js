import React from 'react';
import PropTypes from 'prop-types';
import { Card, Button, Icon } from 'antd';
import Markdown from 'react-markdown';

export default function JobListingContainer({ job, closeListing }) {
  const title = (
    <div>
      <div>
        {`${job.title} - ${job.company} - ${job.location}`}
      </div>
      <div>
        <a href={job.link} target="_blank" rel="noopener noreferrer">
          <Button type="primary">Apply Now</Button>
        </a>
      </div>
    </div>
  );

  return (
    <Card title={title} extra={<Icon onClick={() => closeListing()} type="close" />}>
      <Markdown source={job.description} />
    </Card>
  );
}

JobListingContainer.propTypes = { job: PropTypes.object, closeListing: PropTypes.func };
JobListingContainer.defaultProps = { closeListing: f => f };
