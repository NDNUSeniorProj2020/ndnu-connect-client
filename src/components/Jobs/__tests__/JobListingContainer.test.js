import React from 'react';
import { shallow } from 'enzyme';

import JobListingContainer from '../JobListingContainer';

const job = {
  "id": 1,
  "title": "test1",
  "description": "description test",
  "company": "Google",
  "location": "Mountain View, CA",
  "qualifications": "bs degree",
  "pay": "1234",
  "link": "google.com",
  "date": "2020-04-16T12:53:39.783183-07:00",
  "type": "FULL",
  "user": 2
};

describe('tests for JobListingContainer', () => {
  describe('snapshot tests', () => {
    it('renders without crashing', () => {
      const wrapper = shallow(<JobListingContainer job={job} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
