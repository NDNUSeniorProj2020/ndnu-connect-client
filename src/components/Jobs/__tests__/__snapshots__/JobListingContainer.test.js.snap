// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for JobListingContainer snapshot tests renders without crashing 1`] = `ShallowWrapper {}`;
